From dcf48dbd1c18124f3d1aa0e4f41a83501109d1be Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:19:38 -0400
Subject: [alsa] usb-audio: Use SNDRV_DEV_CODEC for mixer objects

Message-id: <140972823573-118-patchset-jkysela@redhat.com>
Patchwork-id: 90818
O-Subject: [RHEL 7.1 PATCH 117/442] ALSA: usb-audio: Use SNDRV_DEV_CODEC for mixer objects
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Instead of SNDRV_DEV_LOWLEVEL, use SNDRV_DEV_CODEC type for mixer
objects so that they are managed in a proper release order.
No functional change at this point.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 9cbb2808ccfb3c64300151baf73bd5f11feab77f)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/usb/mixer.c b/sound/usb/mixer.c
index aa9bc19..8c152b0 100644
--- a/sound/usb/mixer.c
+++ b/sound/usb/mixer.c
@@ -2373,7 +2373,7 @@ int snd_usb_create_mixer(struct snd_usb_audio *chip, int ctrlif,
 
  snd_usb_mixer_apply_create_quirk(mixer);
 
- err = snd_device_new(chip->card, SNDRV_DEV_LOWLEVEL, mixer, &dev_ops);
+ err = snd_device_new(chip->card, SNDRV_DEV_CODEC, mixer, &dev_ops);
  if (err < 0)
   goto _error;
 
-- 
1.7.1